design.md

```diff
@@ -1,13 +1,50 @@
 # Design
+-- External Language -- 
+
+Lets make "Picobot" more robotic (in a thematic way). My design for this is to use robotic language but not make it feel like a programming language 
+
+Here is an example of what that could look like:
+```
+Engage Directive A:
+
+Detecting Right not Left Directive A
+Transition Up via Directive A
+
+Detecting Up Left Directive B
+Transitioning Right via Directive B
+```
+This language is based on Detecting a "state of existance" which is not just the available locations to move but also the current directive. 
+
+In this language "states" are defined by "Directives." Intuitionally, this version of the syntax makes each directive (state) something that needs to be detected just like the blocks around the Picobot. 
+
+Once detected, the result is the transition (maybe navigate or Move? whats a roboty word for move). It states a direction to move and what Directive to continue with. This uses the word "via" which I like because (1) its short and (2) its very intuitive. You are moving in a direction under a new set of rules. 
+
+Engage is then the method of starting this code at a specific state. State can be any string cause why not. Also maybe Detecting Right Left, means that *you are free to move to the right and the left)* rather then there are blocks there. Might be a bit more intuitive? 
```

Comment:
I like this way of thinking about block detection. It makes a lot of sense but is completely different than the original picobot language.